sim/warlock/inferno.go

```diff
@@ -0,0 +1,203 @@
+
+			// TODO: account for fire spell damage
+			return stats.Stats{
+				stats.Stamina:          ownerStats[stats.Stamina] * 0.75,
+				stats.Intellect:        ownerStats[stats.Intellect] * 0.3,
+				stats.Armor:            ownerStats[stats.Armor] * 0.35,
+				stats.AttackPower:      ownerStats[stats.SpellPower] * 0.57,
+				stats.SpellPower:       ownerStats[stats.SpellPower] * 0.15,
+				stats.SpellPenetration: ownerStats[stats.SpellPenetration],
+				stats.MeleeHit:         ownerHitChance * core.MeleeHitRatingPerHitChance,
+				stats.SpellHit:         ownerHitChance * core.SpellHitRatingPerHitChance,
+				stats.Expertise: (ownerStats[stats.SpellHit] / core.SpellHitRatingPerHitChance) *
+					PetExpertiseScale * core.ExpertisePerQuarterPercentReduction,
+			}
+		}
+
+	infernal := &InfernalPet{
+		Pet: core.NewPet(
+			"Infernal",
+			&warlock.Character,
+			stats.Stats{
+				stats.Strength:  331,
+				stats.Agility:   113,
+				stats.Stamina:   361,
+				stats.Intellect: 65,
+				stats.Spirit:    109,
+				stats.Mana:      0,
+				stats.MeleeCrit: 3.192 * core.CritRatingPerCritChance,
+			},
+			statInheritance,
+			false,
+			false,
+		),
+		owner: warlock,
+	}
+
+	infernal.AddStatDependency(stats.Strength, stats.AttackPower, 2)
+	infernal.AddStat(stats.AttackPower, -20)
+
+	// infernal is classified as a warrior class, so we assume it gets the
+	// same agi crit coefficient
+	infernal.AddStatDependency(stats.Agility, stats.MeleeCrit, core.CritRatingPerCritChance*1/62.5)
+
+	// command doesn't apply to infernal
+	if warlock.Race == proto.Race_RaceOrc {
+		infernal.PseudoStats.DamageDealtMultiplier /= 1.05
+	}
+
+	infernal.EnableAutoAttacks(infernal, core.AutoAttackOptions{
+		MainHand: core.Weapon{
+			BaseDamageMin:  330,
+			BaseDamageMax:  494.9,
+			SwingSpeed:     2,
+			SwingDuration:  time.Second * 2,
+			CritMultiplier: 2,
+		},
+		AutoSwingMelee: true,
+	})
+	infernal.AutoAttacks.MHConfig.DamageMultiplier *= 3.2
+
+	core.ApplyPetConsumeEffects(&infernal.Character, warlock.Consumes)
+
+	warlock.AddPet(infernal)
+
+	return infernal
+}
+
+func (infernal *InfernalPet) GetPet() *core.Pet {
+	return &infernal.Pet
+}
+
+func (infernal *InfernalPet) Initialize() {
+	immolationAuraDot := core.NewDot(core.Dot{
+		Aura: infernal.RegisterAura(core.Aura{
+			Label:    "Immolation",
+			ActionID: core.ActionID{SpellID: 19483},
+		}),
+		NumberOfTicks:       31,
+		TickLength:          time.Second * 2,
+		AffectedByCastSpeed: false,
+		OnTick: func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
+			// TODO: use highest SP amount of all schools
+			// base formula is 25 + (lvl-50)*0.5 * Warlock_SP*0.2
+			// note this scales with the warlocks SP, NOT with the pets
+
+			// we remove all the spirit based sp since immolation aura doesn't benefit from it, see
+			// JamminL/wotlk-classic-bugs#329
+			coef := core.TernaryFloat64(infernal.owner.GlyphOfLifeTapAura.IsActive(), 0.2, 0) +
```

Comment:
Looks like this coeff can be computed outside the OnTick function for better performance.